db/c_test.c

```diff
@@ -329,6 +480,70 @@ int main(int argc, char** argv) {
     leveldb_options_set_error_if_exists(options, 1);
   }
 
+  StartPhase("put expiry");
+  {
+      leveldb_keymetadata_t meta;
+      int loop, count;
+
+      count = sizeof(gCheckPut2Data) / sizeof(gCheckPut2Data[0]);
+
+      for (loop=0; loop<count; ++loop)
+      {
+          size_t klen, vlen;
+          leveldb_keymetadata_t meta;
+          struct CheckPut2Data * test;
+
+          test=&gCheckPut2Data[loop];
+          klen=strlen(test->m_Key);
+          vlen=strlen(test->m_Value);
+          meta.rep.m_Type=test->m_Type;
+          meta.rep.m_Expiry=test->m_Expiry;
+
+          leveldb_put2(db, woptions, test->m_Key, klen,
+                       test->m_Value, vlen, &err,
+                       &meta);
+          CheckNoError(err);
+      }   // for
+
+      // testing memtable right now
+      for (loop=0; loop<count; ++loop)
+      {
+          size_t klen, vlen;
+          leveldb_keymetadata_t meta;
+          struct CheckPut2Data * test;
+
+          test=&gCheckPut2Data[loop];
+          klen=strlen(test->m_Key);
+          vlen=strlen(test->m_Value);
+
+          CheckGet2(db, roptions, test->m_Key, test->m_Value,
+                    test->m_Type, test->m_Expiry);
+      }   // for
+
+      // close and open to force memory table into .sst upon open
+      leveldb_close(db);
+      leveldb_options_set_error_if_exists(options, 0);
+      db = leveldb_open(options, dbname, &err);
+      CheckNoError(err);
+
+      // now testing get from a level-0 .sst file
```

Comment:
Would be good if we could force compactions and check other levels, too. Not sure if this is feasible.